transport/processing/processing_component.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+import sys
+import yaml
+
+import gci.componentmodel as cm
+import processing.config as config
+import processing.processing as processing
+
+import ci.util
+import container
+import container.registry
+import product.v2
+
+
+def parse_component_descriptor(path: str):
+    with open(path) as desc_file:
+        return cm.ComponentDescriptor.from_dict(yaml.safe_load(desc_file))
+
+
+def _download_descriptor(name: str, version: str, ctx_base_url: str):
+    try:
+        return product.v2.download_component_descriptor_v2(
+                component_name=name,
+                component_version=version,
+                ctx_repo_base_url=ctx_base_url,
+        )
+    except container.registry.OciImageNotFoundException as err_not_found:
+        ci.util.error(err_not_found)
+        sys.exit(1)
+
+
+def _resolve_cd_from_references(component_descriptor: cm.ComponentDescriptor):
+    ctx_base_url = component_descriptor.component.repositoryContexts[0].baseUrl
+
+    def __enumerate_cd_from_references(
+            component_descriptor: cm.ComponentDescriptor,
+            ctx_base_url: str
+        ):
+        for comp_ref in component_descriptor.component.componentReferences:
+            oci_ref = product.v2._target_oci_ref_from_ctx_base_url(
+                component_name=comp_ref.componentName,
+                component_version=comp_ref.version,
+                ctx_repo_base_url=ctx_base_url,
+            )
+            descriptor_path = ComponentTool.gen_yaml_file_path(oci_ref)
+            if os.path.isfile(descriptor_path):
+                yield parse_component_descriptor(descriptor_path)
+            else:
+                yield _download_descriptor(comp_ref.componentName, comp_ref.version, ctx_base_url)
+
+    for cd_from_ref in __enumerate_cd_from_references(component_descriptor, ctx_base_url):
+        yield (cd_from_ref.component.name, cd_from_ref)
+        # Recurse if a component descriptor has itself other references
+        if cd_from_ref.component.componentReferences != []:
```

Comment:
nit: I think relying on python's boolean-conversions would be more pythonic here (`if  cd_from_ref.component.componentReferences: ...`)